analysis/quantifier-comparisons/13-spatial-transcriptomics-benchmarking.Rmd

```diff
@@ -504,23 +651,59 @@ gene_detect_df <- rowdata_df_filtered %>%
 
 ggplot(gene_detect_df, aes(x = tools_detected)) +
   geom_bar() +
-  scale_x_upset(n_intersections = 3)
+  scale_x_upset(n_intersections = 4)
+```
+
+For the most part genes are found in all tools, however it looks like there is a chunk of genes that we would miss that are found in Spaceranger that are not found in either of the Alevin-fry tools. 
+Are any of those genes involved in important pathways that we might want to make sure that we don't lose? 
+
+```{r}
+# spaceranger only genes to do ORA 
+spaceranger_only_genes <- gene_detect_df %>%
+  dplyr::filter(tools_detected == "spaceranger") %>%
+  dplyr::select(gene_id) %>%
+  dplyr::left_join(gene_symbols_df, by= "gene_id") %>%
+  tidyr::drop_na() %>%
+  unique() %>%
+  dplyr::pull(symbol)
+
+spaceranger_only_genes
+```
+
+```{r}
+spaceranger_ora_results <- enrichGO(gene = spaceranger_only_genes,
+                           universe = background_genes,
+                           keyType = "SYMBOL",
+                           OrgDb = org.Hs.eg.db,
+                           ont = "BP",
+                           pAdjustMethod = "BH",
+                           pvalueCutoff = 0.00001)
 ```
 
+
+```{r}
+# look at gene ontology results 
+spaceranger_go_results <- spaceranger_ora_results@result %>%
+  as.data.frame() %>%
+  dplyr::filter(p.adjust < 0.2)
+spaceranger_go_results
+```
+
+The genes that are identified in only Spaceranger are not enriched for any specific pathway and also appear to be comprised of a lot of ribosomal genes, mitochondrial genes, and long intergenic non-coding RNA's. 
```

Comment:
Is this based mostly on looking at the list? Is there a way we could quantify this easily? If we have a table where the gene type is a column, could we summarize it for a more quantitative look?

Reply:
After some thinking about this, I went ahead and merged the list of genes with the gtf file and obtained the `gene_biotype`. I added a graph to count the type of genes that are lost and sorted them based on frequency, and the majority of genes that are lost come from long non coding RNA's or pseudogenes with protein coding genes making up the third highest group. Let me know what you think about this approach. 
<img width="771" alt="Screen Shot 2021-12-15 at 4 21 33 PM" src="https://user-images.githubusercontent.com/54039191/146274012-993c122d-a892-4741-b07c-4aa78ce9b5bd.png">